test/extended/syncer/syncer.go

```diff
@@ -0,0 +1,79 @@
+package syncer
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	apb "github.com/kcp-dev/kcp-tests/test/extended/apibinding"
+	exutil "github.com/kcp-dev/kcp-tests/test/extended/util"
+)
+
+var _ = g.Describe("[area/transparent-multi-cluster]", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		k = exutil.NewCLIWithWorkSpace("kcp-syncer")
+	)
+
+	g.It("Author:pewang-Critical-[Smoke][BYO] Validate create, modifying, deleting a deployment from KCP gets synced to the pcluster", func() {
+		pclusterKubeconfig := os.Getenv("PCLUSTER_KUBECONFIG")
+		if pclusterKubeconfig == "" {
+			g.Skip("None pcluster kubeconfig set for the test scenario")
+		}
+		myWs := k.WorkSpace()
+		k.SetGuestKubeconf(pclusterKubeconfig)
+
+		g.By("# Create workload sync and generate syncer resources manifests in current workspace")
+		mySyncer := NewSyncTarget()
+		mySyncer.OutputFilePath = "/tmp/" + myWs.Name + "." + mySyncer.Name + ".yaml"
+		mySyncer.Create(k)
+		defer mySyncer.Clean(k)
+
+		g.By("# Apply syncer resources on pcluster and wait for synctarget become ready")
+		defer k.AsGuestKubeconf().WithoutNamespace().WithoutWorkSpaceServer().Run("delete").Args("-f", mySyncer.OutputFilePath).Execute()
+		err := k.AsGuestKubeconf().WithoutNamespace().WithoutWorkSpaceServer().Run("apply").Args("-f", mySyncer.OutputFilePath).Execute()
+		o.Expect(err).ShouldNot(o.HaveOccurred())
+		mySyncer.WaitUntilReady(k)
+		mySyncer.CheckDisplayAttributes(k)
+
+		g.By("# Create a new workspace and create an APIBinding to attach the compute to the new workspace")
+		k.SetupWorkSpace()
+		tempSlice := strings.Split(myWs.ServerURL, "/")
+		myAPIBinding := apb.NewAPIBinding(apb.SetAPIBindingReferencePath(tempSlice[len(tempSlice)-1]), apb.SetAPIBindingReferenceExportName("kubernetes"))
+		myAPIBinding.Create(k)
+		defer myAPIBinding.Clean(k)
+
+		g.By("# Create workload using the shared compute should work well")
+		myDeployment := exutil.NewDeployment()
+		myDeployment.Create(k)
+		defer myDeployment.Clean(k)
```

Comment:
same comment as above.

Reply:
Hmm，it's antomic action for creating, so it's safe to put the clean after it. In other word, if the resource create failed no need to clean up. I could put it before create for unifying our code logic.

Reply:
Addressed.